app.rb

```diff
@@ -0,0 +1,149 @@
+    @rentals = []
+  end
+
+  def start
+    selecting
+  end
+
+  def books_list
+    puts 'No books available!!' if @books.empty?
+    puts "There are #{@books.length} books available"
+
+    @books.each_with_index do |book, index|
+      puts "#{index + 1}) Title: \"#{book.title}\" | Author: #{book.author}"
+    end
+  end
+
+  def peoples_list
+    puts 'No added people!! Please Add new person.' if @people.empty?
+    puts "#{@people.length} peoples added."
+
+    @people.each_with_index do |person, index|
+      puts "#{index + 1})[#{person.class}] Name: #{person.name} | Age: #{person.age} | ID: #{person.id}"
+    end
+  end
+
+  def create_person
+    puts 'Would like to create student (1) or teacher (2)'
+    choice = gets.chomp
+    puts 'Name:'
+    name = gets.chomp
+    puts 'Age:'
+    age = gets.chomp.to_i
+
+    case choice
+    when '1'
+      create_student(age: age, name: name)
+    when '2'
+      create_teacher(age: age, name: name)
+    else
+      puts 'Invalid. Please try again!'
+    end
+  end
+
+  def create_student(age:, name:)
+    puts 'Parent permission [Y/N]'
+    choice = gets.chomp.downcase
+
+    case choice
+    when 'y'
+      student = Student.new(age: age, name: name, parent_permission: true)
+    when 'n'
+      student = Student.new(age: age, name: name, parent_permission: false)
+    else
+      puts 'Invalid. Try again'
+    end
+    @people << student
+    puts 'Student created successfully!!'
+  end
+
+  def create_teacher(age:, name:)
+    puts 'Please add your specialization:'
+    specialization = gets.chomp
+    teacher = Teacher.new(age: age, name: name, specialization: specialization)
+    @people << teacher
+    puts 'Teacher created successfully!!'
+  end
+
+  def create_book
+    puts 'Title:'
+    title = gets.chomp
+    puts 'Author:'
+    author = gets.chomp
+
+    if title != '' && author != ''
+      book = Book.new(title, author)
+      @books << book
+      puts 'Book created successfully!!'
+    else
+      puts 'Please Enter book title and author'
+    end
+  end
+
+  def new_rental # rubocop:todo Metrics/MethodLength
+    if @books.empty? && @people.empty?
+      puts 'No books and People available'
+    else
+      puts 'Choose the number of the book you want to rent:'
+      @books.each_with_index do |book, index|
+        puts "#{index + 1}: Title: #{book.title} | Author: #{book.author}"
+      end
+      book_id = gets.chomp.to_i - 1
+      choosen_book = @books[book_id]
+
+      puts 'Type your ID:'
+      @people.each do |person|
+        puts "[#{person.class}] Name: #{person.name} | Age: #{person.age} | ID: #{person.id}"
+      end
+      renter_id = gets.chomp.to_i
+      renter = @people.select { |person| person.id == renter_id }.first
+
+      print 'Enter the date[yyyy-mm-dd]: '
+      date = gets.chomp.to_s
+      rental = Rental.new(date: date, person: renter, book: choosen_book)
+      @rentals << rental
+      puts 'Book rented successfully!!'
+    end
+  end

```

Comment:
- [ ]  The app crashes with the attached error message when I tried creating a rental. Kindly fix this to make your app work as expected.

<img width="476" alt="chrome_08aQLgdo5B" src="https://user-images.githubusercontent.com/81468675/175522413-bacb7fca-1af3-4072-9897-2700d5e5c6ec.png">

Reply:
- [ ] I am getting an error when trying to create the rental kindly fix it.


![image](https://user-images.githubusercontent.com/38635158/175652560-605ec09e-b998-4600-9346-ddb7d847743a.png)